### PR TITLE
Change to Target URL Format

### PR DESCRIPTION
**Pull request by Johnng007**
(Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.)

What does this implement/fix? Explain your changes.
---------------------------------------------------
````
This fixes an issue with resolving IPs to hostnames using the conf.socket.gethostbyname(full_host) snippet.
Albeit i feel this task is not necessary as most hardened environments (like in OSCP & CTFs) purposely do not allow DNS resolving, users within a VPN environment would still experience this issue.

Owing to this i would like to make the following changes 
1. In the input for asset to scan users should enter only IP Address or Hostname without http://
2. Comment out full_ip = conf.socket.gethostbyname(full_host) in fullscan.py
3. Change nmap {full_ip} to  {full_host}
4. For nikto and dirsearch before {full_host} include http:// just to be http://{full_host}
````
…

Does this close any currently open issues?
------------------------------------------
````
Yes
````
…

Any relevant logs, error output, etc?
-------------------------------------
````
Traceback (most recent call last): File "/usr/local/bin/webmap", line 56, in <module> main() File "/usr/local/bin/webmap", line 48, in main conf.call_def(conf.full_scan) File "/home/kali/Desktop/TOOLS/WEB ATTACK/WebMap/conf/conf.py", line 80, in call_def func() File "/home/kali/Desktop/TOOLS/WEB ATTACK/WebMap/modules/fullscan.py", line 31, in full_scan full_ip = conf.socket.gethostbyname(full_host) socket.gaierror: [Errno -2] Name or service not known
````
…

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Operating System: kali Linux** …

**Platform:Kali Linux Python 3** …

**Target Platform: Ubuntu** …

**Python Version: Python 3** …

### CODE REVIEW

File: modules/fullscan.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            "green",
+            attrs=["bold"],
+        ))
+
+    conf.not_valid(full_scan, full_host)
+    full_output = conf.dir_output(full_output, "reports/All", full_host)
+
+    conf.create_dir(full_output)
+
+    full_ip = conf.socket.gethostbyname(full_host)
```
Comment (Johnng007) [inline modules/fullscan.py:31]: #commenting out
#full_ip = conf.socket.gethostbyname(full_host)

File: modules/fullscan.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    if len(full_host) == 0:
+        conf.clear()
+
+        print("Not Valid Choice Try again")
+        conf.re_open()
+
+        conf.full_host = None
+    elif gnome_installed:
+        conf.os.system(
+            f"gnome-terminal -- bash -c 'nmap -A {full_ip} -o \"{full_output}/nmap.txt\" && bash'"
```
Comment (Johnng007) [inline modules/fullscan.py:51]: #Going with Full Host Instead
f"gnome-terminal -- bash -c 'nmap -A {full_host} -o \"{full_output}/nmap.txt\" && bash'"

File: modules/fullscan.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        )
+        conf.clear()
+
+        conf.os.system(
+            f"gnome-terminal -- bash -c 'python3 {conf.home}/.local/share/dirsearch/dirsearch.py -u {full_host} --simple-report=\"{full_output}/dirsearch.txt\" && bash'"
```
Comment (Johnng007) [inline modules/fullscan.py:56]: #Adding http in input
f"gnome-terminal -- bash -c 'python3 {conf.home}/.local/share/dirsearch/dirsearch.py -u http://{full_host} --simple-report=\"{full_output}/dirsearch.txt\" && bash'"

File: modules/fullscan.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        )
+        conf.clear()
+
+        conf.os.system(
+            f"gnome-terminal -- bash -c 'nikto +h {full_host} -output \"{full_output}/nikto.txt\" && bash'"
```
Comment (Johnng007) [inline modules/fullscan.py:61]: #Adding http in input
f"gnome-terminal -- bash -c 'nikto +h http://{full_host} -output \"{full_output}/nikto.txt\" && bash'"

File: modules/fullscan.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        )
+        conf.clear()
+
+    else:
+        conf.os.system(f"nmap -A {full_ip} -o {full_output}/nmap.txt")
```
Comment (Johnng007) [inline modules/fullscan.py:66]: #Going with full_host instead
conf.os.system(f"nmap -A {full_host} -o {full_output}/nmap.txt")

File: modules/fullscan.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+        conf.os.system(
+            f"python3 ~/.local/share/dirsearch/dirsearch.py -u {full_host} --simple-report='{full_output}/dirsearch.txt'"
+        )
+
+        conf.os.system(f"nikto +h {full_host} -output {full_output}/nikto.txt")
```
Comment (Johnng007) [inline modules/fullscan.py:72]: #Going with full_host instead
conf.os.system(f"nikto +h http://{full_host} -output {full_output}/nikto.txt")

File: modules/fullscan.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+import conf.conf as conf
+
+
+def full_scan():
+    print("===========================================================")
+    print(conf.colored(conf.text2art("All The Scans", "small"), "cyan"))
+    print("===========================================================")
+
+    full_host = input(
+        conf.colored("\nEnter the target URL : ", "green", attrs=["bold"]))
```
Comment (Johnng007) [inline modules/fullscan.py:18]: #Change request parameter
conf.colored("\nEnter the target IP, Hostname or FQDN : ", "green", attrs=["bold"]))

**Review by Johnng007 — COMMENTED** [review]
Resolving inability to scan IP based URLs

**Comment by Anteste** [comment]
Hey John , thanks for the submiting this pull request.
Sorry if I didn't answer u earlier
You did a Good Job ;)

**Pull request merged**